docs/updating/update-to-40.md

```diff
@@ -14,6 +14,65 @@ modified_at: 2023-11-13
+
+{@img assets/img/image_insert_dropdown.png 772 Image insert dropdown in the main editor toolbar.}
+
+You may want to update your toolbar configuration to make use of the updated feature if you did not use it before.
+
+```js
+import { Image, ImageCaption, ImageResize, ImageStyle, ImageToolbar } from '@ckeditor/ckeditor5-image';
+import { LinkImage } from '@ckeditor/ckeditor5-link';
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		plugins: [ Image, ImageToolbar, ImageCaption, ImageStyle, ImageResize, LinkImage ],
+		// the insert image dropdown toolbar item
+		toolbar: [ 'insertImage', /* ... */ ],
+	} )
+	.then( /* ... */ )
+	.catch( /* ... */ );
+```
+
+By default, the function automatically detects all available upload methods. For example, it will detect and add the `ImageInsertViaUrl` if it is enabled. While no configuration is required for this feature, you might want to limit the options available from the image insert dropdown. Learn more about the toolbar dropdown configuration in the {@link features/images-installation#configuring-the-toolbar-dropdown installation guide}.
+
+#### Image upload plugins
+
+These are all available plugins that provide integrations to the insert image dropdown:
+
+* `ImageUpload` &ndash; provides the default upload from computer toolbar button
+* `ImageInsertViaUrl` &ndash; provides the insert image via URL integration and toolbar item
+* `ImageInsert` &ndash; delivers both of the integrations mentioned above:
+	* `ImageUpload`
+	* `ImageInsertViaUrl`
+
+Additional image inserting plugins:
+* `CKBox` &ndash; asset manager that provides image upload and editing capabilities
+* `CKFinder` &ndash; image manager
+
+#### New image upload icons
+
+Due to the changes to the image insert mechanisms, new toolbar icons have been introduced and replaced the old {@icon @ckeditor/ckeditor5-core/theme/icons/image.svg Image} image icon.
+
+Now there are:
+* {@icon @ckeditor/ckeditor5-core/theme/icons/image-upload.svg Image upload} image upload icon that is the default for the dropdown or for the upload from computer command
+* {@icon @ckeditor/ckeditor5-core/theme/icons/image-asset-manager.svg Image manager} image manager icon
+* {@icon @ckeditor/ckeditor5-core/theme/icons/image-url.svg Insert via URL} insert via URL icon.
+
+If there are more than one insert method configured, the toolbar dropdown will use the {@icon @ckeditor/ckeditor5-core/theme/icons/image-upload.svg Image upload} default icon. If only one image insert method is available, the toolbar will use the respective icon instead.
```

Comment:
As far as I remember, it will load the icon of the next integration according to the order. So if I have no upload, but asset + URL, the icon will the asset one.

Reply:
In commit.